### PR TITLE
Add code for set operators, and update existing doc

### LINKED ISSUE #25721: Add content and code samples for new LINQ operators
Add content and code samples for the newly added LINQ operators:

- [ ] `Enumerable.Zip`
- [ ] `Enumerable.ChunckBy`
- [ ] `SingleOrDefault`, `FirstOrDefault`, and `LastOrDefault` - overloads that allow specifying a default
- [ ] Updates to existing APIs exposing C# 8 index and range
- [ ] `Enumerable.*By` additions:
  - `DistinctBy`
  - `ExceptBy`
  - `IntersectBy`
  - `UnionBy`
  - `MinBy`
  - `MaxBy`

Related to the various new LINQ APIs captured in the runtime PR: https://github.com/dotnet/runtime/issues/47231

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Add code for set operators, and update existing doc

Fixes #25721

[🔗 Internal preview: Set operations (C#)](https://review.docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/set-operations?branch=pr-en-us-25979)

### CODE REVIEW

**Comment by BillWagner** [comment]
I gave this a full read, and everything looks good, once the project build issues are addressed.

**Comment by adegeo** [comment]
Passing build error to @BillWagner as it seems the LocateProjects tool is the culprit.

File: docs/csharp/programming-guide/concepts/linq/snippets/set-operators/set-operators.csproj [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (BillWagner) [inline docs/csharp/programming-guide/concepts/linq/snippets/set-operators/set-operators.csproj:5]: Add this line to the linq.csproj in the parent folder.

File: docs/csharp/programming-guide/concepts/linq/snippets/set-operators/Program.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿using SolarSystem;
+
+Program.DistinctByExample();
+Program.ExceptByExample();
+Program.IntersectByExample();
+Program.UnionByExample();
```
Comment (BillWagner) [inline docs/csharp/programming-guide/concepts/linq/snippets/set-operators/Program.cs:6]: These four statements could move into the Program.cs file in the parent folder. (Bonus points if you want to update that program to use top level statements).

**Review by BillWagner — CHANGES_REQUESTED** [review]
I figured out why the snippets build is failing. The CI build is working correctly, but because you've added a folder with a project, the project in the parent folder will now fail. That's because it will now see a Main method, and top level statements. The parent folder also uses .NET Core 3.1.

I suggest merging the code into one project (subfolders are fine), and then it will work.

File: docs/csharp/programming-guide/concepts/linq/snippets/set-operators/set-operators.csproj (hunk 1 above)
Comment (IEvangelist) [reply docs/csharp/programming-guide/concepts/linq/snippets/set-operators/set-operators.csproj:5]: I kept it as two projects, moved the one and added a solution. All are targeting .NET 6

**Review by BillWagner — APPROVED** [review]

**Pull request merged**